python/sdist/setup_clibs.py

```diff
@@ -43,23 +43,17 @@ def get_sundials_sources() -> List[str]:
         os.path.join('src', 'idas', 'idas_nls_sim.c'),
         os.path.join('src', 'idas', 'idaa_io.c'),
         os.path.join('src', 'sundials', 'sundials_math.c'),
-        os.path.join('src', 'sundials', 'sundials_mpi.c'),
-        os.path.join('src', 'sundials', 'sundials_sptfqmr.c'),
         os.path.join('src', 'sundials', 'sundials_matrix.c'),
-        os.path.join('src', 'sundials', 'sundials_pcg.c'),
         os.path.join('src', 'sundials', 'sundials_direct.c'),
-        os.path.join('src', 'sundials', 'sundials_spgmr.c'),
-        os.path.join('src', 'sundials', 'sundials_spbcgs.c'),
         os.path.join('src', 'sundials', 'sundials_nvector_senswrapper.c'),
         os.path.join('src', 'sundials', 'sundials_dense.c'),
         os.path.join('src', 'sundials', 'sundials_nvector.c'),
         os.path.join('src', 'sundials', 'sundials_version.c'),
-        os.path.join('src', 'sundials', 'sundials_spfgmr.c'),
-        os.path.join('src', 'sundials', 'sundials_sparse.c'),
         os.path.join('src', 'sundials', 'sundials_iterative.c'),
         os.path.join('src', 'sundials', 'sundials_nonlinearsolver.c'),
         os.path.join('src', 'sundials', 'sundials_linearsolver.c'),
         os.path.join('src', 'sundials', 'sundials_band.c'),
+        os.path.join('src', 'sundials', 'sundials_futils.c'),
```

Comment:
I guess, changes from python/sdist/setup_clibs.py will need to be applied to matlab compilation functions as well. Maybe at some point introduce a list of source files that can be used from both python and matlab...

Reply:
good point!